src/transaction.js

```diff
@@ -39,43 +39,61 @@ export class Transaction {
     }
     this._networkPassphrase = networkPassphrase;
 
-    const txe = envelope.value();
+    const txEnvelope = envelope.value();
     let sourceAccount;
     this._envelopeType = envelope.switch();
     switch (this._envelopeType) {
       case xdr.EnvelopeType.envelopeTypeTxV0():
-        sourceAccount = txe.tx().sourceAccountEd25519();
+        sourceAccount = txEnvelope.tx().sourceAccountEd25519();
         break;
       case xdr.EnvelopeType.envelopeTypeTx():
-        sourceAccount = txe
+        sourceAccount = txEnvelope
           .tx()
           .sourceAccount()
           .ed25519();
         break;
+      case xdr.EnvelopeType.envelopeTypeTxFeeBump():
+        sourceAccount = txEnvelope
+          .tx()
+          .feeSource()
+          .ed25519();
+        break;
       default:
-        // TODO: Add support for FeeBumpTransaction
-        throw new Error('Invalid transaction.');
+        throw new Error('Unknown TransactionEnvelope');
     }
 
     // since this transaction is immutable, save the tx
-    this.tx = txe.tx();
-    this.source = StrKey.encodeEd25519PublicKey(sourceAccount);
-    this.fee = this.tx.fee();
-    this._memo = this.tx.memo();
-    this.sequence = this.tx.seqNum().toString();
+    this.tx = txEnvelope.tx();
+    let tx = this.tx;
 
-    const timeBounds = this.tx.timeBounds();
+    // if isFeeBumpTx override tx with innerTx, which we use to expose data about the innerTx
+    if (this.isFeeBump()) {
+      this.fee = this.tx.fee().toString();
+      tx = tx
+        .innerTx()
+        .value()
+        .tx();
+      this.source = StrKey.encodeEd25519PublicKey(tx.sourceAccount().ed25519());
+      this.feeSource = StrKey.encodeEd25519PublicKey(sourceAccount);
+    } else {
+      this.source = StrKey.encodeEd25519PublicKey(sourceAccount);
+      this.fee = tx.fee();
+    }
+
+    this._memo = tx.memo();
+    this.sequence = tx.seqNum().toString();
```

Comment:
It's hard to see what's the difference between a feebump and non-feebump. I think it would be more readable (and only slightly less efficient) to initialize the fields inconditionally as a non-feebump and amend them if it's a feebump.

Reply:
@2opremio good point!